core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -491,6 +491,28 @@ private static String toSql(RelNode root, SqlDialect dialect,
     relFn(relFn).ok(expected);
   }
 
+  @Test void testSelectWhereIn2() {
```

Comment:
how are these tests related to the issue solved?

Reply:
I am guessing that the IN list is converted to a union and it's converted back to a values
Perhaps you can add a comment to say that

Reply:
I had the same idea before. I added this test case because I want to know if RelBuilder generates the UNION when IN-List has CAST. But It doesn't.